### ISSUE TITLE
[f2s_child_worker::ERROR] Bad fast5

### ISSUE THREAD

**Issue by MeHelmy** [hasindu2008/slow5tools#83]
I am running the command as follows:

`slow5tools f2s ${dir_name}/fast5 -d "${outDir}"`
Where `${dir_name}` is the input directory and `${outDir}` the output directory.

But shortly, it gives me this error:

```
list_all_items] Looking for '*.fast5' files in dir_name/fast5
[f2s_main] 3245 fast5 files found - took 0.391s
[f2s_main] Just before forking, peak RAM = 0.000 GB
[f2s_iop] 8 proceses will be used.
[f2s_iop] Spawning 8 I/O processes to circumvent HDF hell.
[f2s_child_worker::ERROR] Bad fast5: Fast5 file 'dir_name/fast5/P5a_279.fast5' could not be opened or is corrupted.
[f2s_iop] Child process 1969843 exited with status=1.

```

I appreciate your feedback/suggestions,
Medhat

**Comment by hasindu2008** [comment]
Could you isolate the file 'dir_name/fast5/P5a_279.fast5' which is causing the problem in to a separate directory and see if it really corrupted by trying to run Guppy on it? 
Alternatively you can share this file so that I could investigate.

**Comment by clock-matt** [comment]
I had the same error come up today. I'll try and run guppy on it and get back to you

**Comment by clock-matt** [comment]
Also, this was my error log: 
[list_all_items] Looking for '*.fast5' files in data/YM1
[list_all_items] Looking for '*.fast5' files in data/YM1/YM1
[list_all_items] Looking for '*.fast5' files in data/YM1/YM1/20220427_1557_MN35906_FAT17286_8bc2484f
[list_all_items] Looking for '*.fast5' files in data/YM1/YM1/20220427_1557_MN35906_FAT17286_8bc2484f/fast5
[list_all_items] Looking for '*.fast5' files in data/YM1/YM1/20220427_1557_MN35906_FAT17286_8bc2484f/other_reports
[f2s_main] 200 fast5 files found - took 0.000s
[f2s_main] Just before forking, peak RAM = 0.000 GB
[f2s_iop] 8 proceses will be used.
[f2s_iop] Spawning 8 I/O processes to circumvent HDF hell.
[f2s_child_worker::ERROR]^[[1;31m Bad fast5: Fast5 file 'data/YM1/YM1/20220427_1557_MN35906_FAT17286_8bc2484f/fast5/FAT17286_75202811_56.fast5' could not be opened or is corrupted.^[[0m
[f2s_child_worker::ERROR]^[[1;31m Bad fast5: Fast5 file 'data/YM1/YM1/20220427_1557_MN35906_FAT17286_8bc2484f/fast5/FAT17286_75202811_248.fast5' could not be opened or is corrupted.^[[0m
[f2s_child_worker::ERROR]^[[1;31m Bad fast5: Fast5 file 'data/YM1/YM1/20220427_1557_MN35906_FAT17286_8bc2484f/fast5/FAT17286_75202811_39.fast5' could not be opened or is corrupted.^[[0m
[f2s_child_worker::INFO]^[[1;34m Summary - total fast5: 25, bad fast5: 0^[[0m
[f2s_child_worker::INFO]^[[1;34m Summary - total fast5: 25, bad fast5: 0^[[0m
[f2s_child_worker::INFO]^[[1;34m Summary - total fast5: 25, bad fast5: 0^[[0m
[f2s_child_worker::INFO]^[[1;34m Summary - total fast5: 25, bad fast5: 0^[[0m
[f2s_child_worker::INFO]^[[1;34m Summary - total fast5: 25, bad fast5: 0^[[0m
[f2s_iop] Child process 177782 exited with status=1.

**Comment by clock-matt** [comment]
I did get a chance to try them with guppy and guppy also couldn't load them. I'll have to dig through them to see what might be going on. 

Do you know if the worker would have continued to process the other fast5 files, or does it stop after the error?

**Comment by hiruna72** [comment]
Hello @clock-matt,

Thank you reporting this. 

At the moment, if `f2s` comes across a bad fast5 file it errors out. Its partial output should be discarded. 
The best method is to identify and filter out the bad fast5 files before running `f2s`. This can be tricky.

We are delivering the "best effort" `f2s` to convert each and every fast5 file that can be basecalled through guppy.

As you have figured out if guppy also fails this is a serious issue with the fast5 file itself. Can you please share one of your bad fast5 files? It might be a good idea to report the file to ONT as well.

Thank you.

**Comment by hasindu2008** [comment]
@clock-matt 

If you could move the three files `data/YM1/YM1/20220427_1557_MN35906_FAT17286_8bc2484f/fast5/FAT17286_75202811_56.fast5'` , `data/YM1/YM1/20220427_1557_MN35906_FAT17286_8bc2484f/fast5/FAT17286_75202811_248.fast5` and `data/YM1/YM1/20220427_1557_MN35906_FAT17286_8bc2484f/fast5/FAT17286_75202811_39.fast5'  into a separate directory outside and call f2s it is likely that I will work (assuming only those 3 files are corrupted).  

When I was converting ~500 datasets, came across a handful of samples where two or three fast5 files were corrupted. I moved those files and recalled f2s. You could  try the same.

**Comment by hasindu2008** [comment]
@MeHelmy 
Any update on your side?

**Comment by clock-matt** [comment]
I tried converting them from their own directory and had the same error.

**Comment by clock-matt** [comment]
[FAT17286_75202811_39.fast5.gz](https://github.com/hasindu2008/slow5tools/files/9737656/FAT17286_75202811_39.fast5.gz)
[FAT17286_75202811_56.fast5.gz](https://github.com/hasindu2
[FAT17286_75202811_248.fast5.gz](https://github.com/hasindu2008/slow5tools/files/9737658/FAT17286_75202811_248.fast5.gz)
008/slow5tools/files/9737657/FAT17286_75202811_56.fast5.gz)

**Comment by hasindu2008** [comment]
@clock-matt the files you attached when extracted seem to be empty files 0KB.
<img width="516" alt="image" src="https://user-images.githubusercontent.com/12987163/194668764-f91d57e5-2080-4859-822d-55da64bf930e.png">

Is this something that happened when you uploading or are your original files also like that?

**Comment by clock-matt** [comment]
Yeah it looks like something went wrong when I was downloading the files originally. So that seems to be the cause of the error for me.
